crates/solvers/src/domain/solver/legacy.rs

```diff
@@ -29,7 +29,8 @@ impl Legacy {
 
     pub async fn solve(&self, auction: auction::Auction) -> Vec<solution::Solution> {
         match self.0.solve(&auction).await {
-            Ok(solution) => vec![solution.with_id(solution::Id(0))],
+            Ok(solution) if !solution.is_empty() => vec![solution.with_id(solution::Id(0))],
+            Ok(_) => vec![],
```

Comment:
I find this harder to read than a good old `if/else` statement inside a single Ok branch

Reply:
Adjusted.